src/mlpack/core/metrics/lmetric.hpp

```diff
@@ -116,6 +116,10 @@ typedef LMetric<2, true> EuclideanDistance;
  */
 typedef LMetric<INT_MAX, false> ChebyshevDistance;
 
+/**
+ * The HDBSCAN metric.
+ */
+typedef LMetric<INT_MAX, true> HdbscanDistance;
```

Comment:
That's there for a reason... the BinarySpaceTree type *requires* the distance to be `LMetric` or some variant thereof because the `MinDistance()` and `MaxDistance()` functions are specialized.  So overloading the distance here will cause the bounds to fail and therefore the results you get are wrong (this is why you have to use naive mode---but you should still check whether it follows the triangle inequality).  You will need to use a different type of tree (like `BallTree` or something else) by default for this to work.